Cesium.CodeGen/Ir/BlockItems/IBlockItem.cs

```diff
@@ -4,8 +4,7 @@ namespace Cesium.CodeGen.Ir.BlockItems;
 
 internal interface IBlockItem
 {
-    bool HasDefiniteReturn => false;
-
+    bool TryUnsafeSubstitute(IBlockItem original, IBlockItem replacement);
```

Comment:
My current understanding is that this way of tree transformation is somewhat fragile and has cascading effects on codebase where changes in signature of methods propagated across whole codebase even if it's relatively simple process change. I was biten by that even after Lower/EmitTo split and it was not pleasant.

My preference to have some Visitor/Rewriter classes like in Roslyn https://github.com/dotnet/roslyn/blob/303412ce74f70f9ea978bba40705cfcb908d901d/src/Compilers/CSharp/Portable/Binder/TypeofBinder.cs#L47

I think this approach will help separate tree transformation processes. @ForNeVeR not so sure about this IIRC.